File: app/src/main/java/com/maxandron/daggerissue325java/bar/BarActivity.java
package com.maxandron.daggerissue325java.bar;

import com.maxandron.daggerissue325java.BaseActivity;
import com.maxandron.daggerissue325java.MyApplication;

import javax.inject.Inject;

public class BarActivity extends BaseActivity {

    @Inject
    Bar mBar;

    @Override
    protected void setupComponent() {
        MyApplication.get(this).getAppComponent().plus(new BarProvider(this)).inject(this);
    }
}

File: app/src/main/java/com/maxandron/daggerissue325java/bar/BarActivity.kt
package com.maxandron.daggerissue325java.bar

import com.maxandron.daggerissue325java.BaseActivity
import com.maxandron.daggerissue325java.MyApplication

import javax.inject.Inject

class BarActivity : BaseActivity() {

    @Inject
    lateinit var mBar: Bar

    override fun setupComponent() {
        MyApplication.get(this).appComponent.plus(BarProvider(this)).inject(this)
    }
}
